packages/frontend/src/creation-form/components/deploy.tsx

```diff
@@ -329,8 +355,53 @@ export const Deploy = ({
         void create();
     }, [devMode, onCreate, onNext, onTx, publicClient, writeAsync]);
 
+    const handleExpirationChange = useCallback(
+        (value: Date) => {
+            const converted = dateToUnixTimestamp(value);
+            setExpirationErrorText(
+                isUnixTimestampInThePast(converted)
+                    ? t("error.expiration.past")
+                    : "",
+            );
+            onStateChange((state) => ({
+                ...state,
+                expiration: converted,
+            }));
+        },
+        [onStateChange, t],
+    );
+
     return (
         <div className="flex flex-col gap-6">
+            <DateTimeInput
+                data-testid="generic-data-step-expiration-input"
+                label={t("general.label.expiration")}
```

Comment:
It'a minor thing but since we moved the input in the deploy step we could rename also the keys in the i18n file? So for example `general.label.expiration` becomes `deploy.label.expiration`.

Reply:
Updated in c17e07b3c2beed1139b16537242c9048d9e2f9db